### ISSUE TITLE
every refresh token fails in OAuth2

### ISSUE THREAD

**Issue by ohayak** [getinsomnia/insomnia#5701]
Hello,
When using OAuth2 tool to fetch automatically tokens for my requests all refresh-tokens attempts failed with the following error [oauth2] Failed to refresh token url=http://localhost:8000/v1/login/token status=200
I tried the same authentication endpoint on Postman seems to work fine. I did some digging in [the code](https://github.com/Kong/insomnia/blob/develop/packages/insomnia/src/network/o-auth-2/get-token.ts) to find this

```javascript
const isSuccessful = statusCode >= 200 && statusCode < 300;
const hasBodyAndIsError = bodyBuffer && statusCode === 400;
if (isSuccessful) {
  if (hasBodyAndIsError) {
    const body = tryToParse(bodyBuffer.toString());
    // If the refresh token was rejected due an oauth2 invalid_grant error, we will
    // return a null access_token to trigger an authentication request to fetch
    // brand new refresh and access tokens.
    if (body?.error === 'invalid_grant') {
      console.log(`[oauth2] Refresh token rejected due to invalid_grant error: ${body.error_description}`);
      const old = await models.oAuth2Token.getOrCreateByParentId(requestId);
      models.oAuth2Token.update(old, transformNewAccessTokenToOauthModel({ access_token: null }));
      return null;
    }
  }

  throw new Error(`[oauth2] Failed to refresh token url=${authentication.accessTokenUrl} status=${statusCode}`);
}
```
in a nutshell, you are throwing an error every time the refresh is successful.
I pushed a fix in this [pull request]

**Comment by filfreire** [comment]
@ohayak thanks for reporting this and submitting a fix. It will go out in an upcoming release
